external-adapter/src/requester.ts

```diff
@@ -82,7 +86,13 @@ export class Requester {
   static errored(jobRunID = '1', error?: AdapterError | Error | string, statusCode = 500) {
     if (error instanceof AdapterError) return error.toJSONResponse()
     if (error instanceof Error)
-      return new AdapterError({ jobRunID, statusCode, cause: error }).toJSONResponse()
+      return new AdapterError({
+        jobRunID,
+        statusCode,
+        message: error.message,
+        name: error.name,
+        cause: error.stack,
```

Comment:
Why did you make this change?

I would keep these two errors separate. The cause and the `AdapterError` that wraps the cause. Here you basically merged it in one. On the output as a user, I would like to see an `AdapterError` with a specific error message but also the ability to investigate the underlying cause.

Reply:
The reason I made this change is that as a user the error cause shows the stack trace from the `AdapterError` but not the underlying Error that would be useful for debugging the underlying cause.

ie. throwing a `new Error("TESTING THROW ERROR")` from inside an EA gives

```
{"jobRunID":"1","status":"errored","statusCode":500,"error":{"name":"AdapterError","message":"TESTING THROW ERROR","stack":"AdapterError: TESTING THROW ERROR\n    at Function.errored (/home/justinkaseman/Desktop/dev/external-adapters-js/external-adapter/dist/requester.js:86:20)\n    at /home/justinkaseman/Desktop/dev/external-adapters-js/bootstrap/dist/index.js:88:94\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","cause":{}}}
```

where as the actual useful stack trace is:

```
Error: TESTING THROW ERROR
    at getBalance (/home/justinkaseman/Desktop/dev/external-adapters-js/blockchain.com/dist/endpoint/balance.js:11:11)
    at /home/justinkaseman/Desktop/dev/external-adapters-js/ea-factories/dist/factories/balance.js:49:34
    at Array.map (<anonymous>)
    at /home/justinkaseman/Desktop/dev/external-adapters-js/ea-factories/dist/factories/balance.js:49:21
    at /home/justinkaseman/Desktop/dev/external-adapters-js/ea-factories/dist/factories/balance.js:71:48
    at Array.flatMap (<anonymous>)
    at /home/justinkaseman/Desktop/dev/external-adapters-js/ea-factories/dist/factories/balance.js:71:29
    at Object.exports.execute (/home/justinkaseman/Desktop/dev/external-adapters-js/blockchain.com/dist/adapter.js:20:58)
    at /home/justinkaseman/Desktop/dev/external-adapters-js/blockchain.com/dist/adapter.js:32:39
    at /home/justinkaseman/Desktop/dev/external-adapters-js/composite/proof-of-reserves/dist/balance.js:40:16 object

```

Reply:
Reverted this